test/gtest/graphapi_capi_mha_backward.cpp

```diff
@@ -0,0 +1,430 @@
+            .InitAndWriteToGPU(handle, dsDescale);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaScaleDS).InitAndWriteToGPU(handle, dsScale);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaScaleDQ).InitAndWriteToGPU(handle, dqScale);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaScaleDK).InitAndWriteToGPU(handle, dkScale);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaScaleDV).InitAndWriteToGPU(handle, dvScale);
+
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaDQ, m_testN, m_testH, m_testS, m_testD);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaDK, m_testN, m_testH, m_testS, m_testD);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaDV, m_testN, m_testH, m_testS, m_testD);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaAmaxDQ);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaAmaxDK);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaAmaxDV);
+        MakeAndAddRealTensorDescriptor(miopenTensorMhaAmaxDS);
+
+        dQDescRef = tensor<T>{m_testN, m_testH, m_testS, m_testD};
+        dKDescRef = tensor<T>{m_testN, m_testH, m_testS, m_testD};
+        dVDescRef = tensor<T>{m_testN, m_testH, m_testS, m_testD};
+
+        test::cpu::MultiHeadAttentionBackwardDataf8(
+            GetTensor<T>(m_realTensorMap[miopenTensorMhaQ]->m_tensorVariant),
+            GetTensor<T>(m_realTensorMap[miopenTensorMhaK]->m_tensorVariant),
+            GetTensor<T>(m_realTensorMap[miopenTensorMhaV]->m_tensorVariant),
+            GetTensor<T>(m_realTensorMap[miopenTensorMhaO]->m_tensorVariant),
+            GetTensor<T>(m_realTensorMap[miopenTensorMhaDO]->m_tensorVariant),
+            softmax,
+            q.mDescale,
+            k.mDescale,
+            v.mDescale,
+            dqScale,
+            dkScale,
+            dvScale,
+            sScale,
+            sDescale,
+            dsScale,
+            dsDescale,
+            oDescale,
+            dO.mDescale,
+            amaxDSRef,
+            amaxDQRef,
+            amaxDKRef,
+            amaxDVRef,
+            dQDescRef,
+            dKDescRef,
+            dVDescRef);
+
+        // get next value for the rest of the tensors (which don't have any particular enum value)
+        GetNextId();
+    }
+
+    virtual void MakeVirtualTensorsAndNodes() override
+    {
+        ////////////// Left part (column) of the Mha backward graph ///////////////
+        ///////////////////////////////////////////////////////////////////////////
+
+        auto mm0 = MakeGapiVirtualTensorDesc(m_testN, m_testH, m_testS, m_testS);
+
+        MakeMatmul(m_realTensorMap[miopenTensorMhaQ]->m_gapiDesc,
+                   m_realTensorMap[miopenTensorMhaK]->m_gapiDesc,
+                   mm0);
+
+        // MakePointwise function will automatically create a tensor for output,
+        // if output is nullptr
+        DescriptorWrapperPtr pwS0;
+        MakePointwise(MIOPEN_POINTWISE_IDENTITY, mm0, nullptr, pwS0, false, m_attentionScale);
```

Comment:
I was asking because in CuDNN's version `attention scale` is an optional node.

Yes, `MUL` node only accepts tensors, and you have to create a tensor `[1, 1, 1, 1]` for a single float value. Broadcasting will propagate the value.

Reply:
If I understand correctly Attention scale is a value not a tensor. Also, its a value in Find 2.0 interface. That's why it was decided to do it  We can discuss all these details on the meeting with Alexander and Amber